quickfixj-core/src/main/java/quickfix/mina/AbstractIoHandler.java

```diff
@@ -145,7 +149,16 @@ public void messageReceived(IoSession ioSession, Object message) throws Exceptio
                 }
             }
         } else {
-            log.error("Disconnecting; received message for unknown session: {}", messageString);
+            boolean logMessageWhenSessionNotFound = true;
+            if (sessionSettings.isSetting(Session.SETTING_LOG_MESSAGE_WHEN_SESSION_NOT_FOUND)) {
+                logMessageWhenSessionNotFound = sessionSettings.getBool(Session.SETTING_LOG_MESSAGE_WHEN_SESSION_NOT_FOUND);
+            }
```

Comment:
Could we move this block to the constructor? The setting is not specific to any session, so we do not need to look it up on every incident.

Reply:
Moved the block, thanks @chrjohn